components/ShoppingList.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+
+const StyledShoppingList = styled.div`
+  max-width: 800px;
+  margin: 0 auto;
+  padding: 20px;
+
+  .items-container {
+    max-height: 400px;
+    overflow-y: auto;
+    border: 1px solid #ccc;
+    padding: 10px;
+  }
+`;
+
+const ItemCard = styled.div`
+  border: 1px solid #ddd;
+  padding: 10px;
+  margin-bottom: 10px;
+  border-radius: 5px;
+  background-color: ${({ category }) => {
+    switch (category) {
+      case "fruits":
+        return "#3bff4b";
+      case "bakery":
+        return "#b06100";
+      case "dairy":
+        return "#7a7fff";
+      case "vegetables":
+        return "#009900";
+      case "meat":
+        return "#FFCDD2";
+      case "beverage":
+        return "#ff7f7f";
+      case "snacks":
+        return "#009900";
+      case "household":
+        return "#d97ae6";
+      case "personal care":
+        return "#d48208";
+      case "other":
+        return "#fc9997";
+      default:
+        return "#E0E0E0"; // Grau als Fallback
+    }
+  }};
+`;
+
+const ShoppingList = () => {
+  const [items, setItems] = useState([]);
+  const [categories, setCategories] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const [itemsResponse, categoriesResponse] = await Promise.all([
+        fetch("/assets/shopping-items.json"),
+        fetch("/assets/categories.json"),
+      ]);
+
+      if (itemsResponse.ok && categoriesResponse.ok) {
+        let itemsData = await itemsResponse.json();
+        const categoriesData = await categoriesResponse.json();
+
+        itemsData = itemsData.map((item, index) => ({
+          ...item,
+          id: `item-${index + 1}`,
+          category: item.category.toLowerCase(),
+        }));
+
+        setItems(itemsData);
+        setCategories(categoriesData);
+      }
+      setLoading(false);
+    };
+
+    fetchData();
+  }, []);
+
+  const getTotalItems = () => {
+    return items.reduce((total, item) => total + item.quantity, 0);
+  };
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <StyledShoppingList>
+      <h1>Shopping List: What I Need to Buy</h1>
+      <p>Total items: {getTotalItems()}</p>
```

Comment:
This change is needed, after changing from the `getTotalItems` function to a derived state

```suggestion
      <p>Total items: {totalItems}</p>
```